test/suites/tracing-tests/test-trace-erc20-xcm.ts

```diff
@@ -123,6 +125,83 @@ describeSuite({
           args: [CHARLETH_ADDRESS],
         })
       ).equals(amountTransferred);
+
+      // Now create a failed XCM transaction by trying to transfer more than available
+      const failedConfig: XcmFragmentConfig = {
+        assets: [
+          {
+            multilocation: {
+              parents: 0,
+              interior: {
+                X1: { PalletInstance: Number(balancesPalletIndex) },
+              },
+            },
+            fungible: 1_700_000_000_000_000n,
+          },
+          {
+            multilocation: {
+              parents: 0,
+              interior: {
+                X2: [
+                  {
+                    PalletInstance: erc20XcmPalletIndex,
+                  },
+                  {
+                    AccountKey20: {
+                      network: null,
+                      key: erc20ContractAddress,
+                    },
+                  },
+                ],
+              },
+            },
+            fungible: amountTransferred * 2n, // Try to transfer twice the available amount
+          },
+        ],
+        beneficiary: CHARLETH_ADDRESS,
+      };
+
+      const failedXcmMessage = new XcmFragment(failedConfig)
+        .withdraw_asset()
+        .clear_origin()
+        .buy_execution()
+        .deposit_asset(2n)
+        .as_v3();
+
+      // Mock the reception of the failed xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: failedXcmMessage,
+      });
+
+      // By calling deployContract() a new block will be created,
+      // including the ethereum xcm transaction + regular ethereum transaction
+      const { contractAddress: eventEmitterAddress_ } = await context.deployContract!(
+        "EventEmitter",
+        {
+          from: alith.address,
+        } as any
+      );
+      eventEmitterAddress = eventEmitterAddress_;
+
+      createTransactionHash = (await context.viem().getBlock()).transactions[0];
```

Comment:
This would probably not work with the runtime fix
```suggestion
      const { contractAddress: eventEmitterAddress_, hash: createTransactionHash_ } = await context.deployContract!(
        "EventEmitter",
        {
          from: alith.address,
        } as any
      );
      eventEmitterAddress = eventEmitterAddress_;
      createTransactionHash = createTransactionHash_;
```

Reply:
It's expected, I added a comment about that:

> IMPORTANT: this test will fail once we will merge https://github.com/moonbeam-foundation/moonbeam/pull/3258

As far as I know, we don't have a way to test latest client against old runtimes in our CI, so I created a temporary test.